designsafe/static/scripts/data-depot/components/data-depot-listing/data-depot-listing.component.js

```diff
@@ -242,6 +243,40 @@ class FilesListingCtrl {
       clearTypeFilters() {
         this.$state.go(this.$state.current, {typeFilters: null}, {reload: true})
       }
+    /**
+     * Check if folder has any python notebooks
+     */
+    hasNotebooks() {
+        if (!this.listing() || window.location.href.includes('agave')) return false;
+        const listing = this.listing().children;
+        const pyNb = listing.filter((item)=> item.ext === 'ipynb');
+        if (_.isEmpty(pyNb)) {
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * Modified version of DataBrowserServicePreviewCtrl.openInJupyter
+     */
+    openInApp(file) {
+        const pathToFile = file.path;
+        let specificLocation = this.$state.current.name;
+        if (specificLocation === 'myData' || specificLocation === 'communityData') {
+            specificLocation = (specificLocation.charAt(0).toUpperCase() + specificLocation.slice(1));
+        } else if (specificLocation.includes('projects')) {
+            const prjNumber = this.DataBrowserService.state().project.value.projectId;
+            specificLocation = 'projects/' + prjNumber;
+        } else if (specificLocation === 'publishedData.view') {
+            specificLocation = 'Published';
+        } 
+        if (file.system === 'designsafe.storage.published') { 
+            specificLocation = 'NHERI-Published';
+        }
+        const fileLocation = `${specificLocation}/${pathToFile}`;
+        const jupyterPath = `http://jupyter.designsafe-ci.org/user/${this.Django.user}/notebooks/${fileLocation}`;
+        return jupyterPath;
+    }
```

Comment:
Is there a way to split this out into a utility function that both components use?

Reply:
Would it be best to move this in to the `FileListing` class or `DataBrowserService`?

Reply:
I think it belongs in `FileListing`, as much as I wish that class didn't even exist.